File: docs/scope/468-reader-text.probe.py
"""#468 probe: every day of the committed QA showcase, served as /api/model-view
serves it, for the Node probe that runs the shipped Episode Log over each day.
Synthetic inputs only: the committed showcase store, opened read-only.

Run from the repo root:
  uv run python docs/scope/468-reader-text.probe.py "$TMPDIR/468-days.json"
  node docs/scope/468-reader-text.probe.mjs "$TMPDIR/468-days.json"
"""
import json
import sys
from datetime import datetime, timedelta
from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]
sys.path.insert(0, str(ROOT))

from ciq_autotune.analyzers.scenario import build_model_view  # noqa: E402
from ciq_autotune.store import Store  # noqa: E402

with Store.open_readonly(str(ROOT / "mockups/qa-e2e.synthetic/harmonic.sqlite")) as store:
    times = store.conn.execute("SELECT MIN(t), MAX(t) FROM cgm_readings").fetchone()
    first, last = (datetime.fromisoformat(t).date() for t in times)
    days, day = [], first
    while day <= last:
        days.append(build_model_view(store, day))
        day += timedelta(days=1)
Path(sys.argv[1]).write_text(json.dumps(days, default=str))
print(f"{len(days)} days, {days[0]['date']} to {days[-1]['date']}")

File: docs/scope/468-reader-text.probe.mjs
// #468 probe: the shipped Episode Log over every showcase day the Python probe
// dumped. Prints each day whose Quiet span covers a Findings row, and how many
// days carry one quiet anchor. Synthetic inputs only.
//
// Run from the repo root: node docs/scope/468-reader-text.probe.mjs "$TMPDIR/468-days.json"
import { readFileSync } from 'node:fs';
import { buildEpisodeLedger, clock } from '../../frontend/day-chart.js';

const days = JSON.parse(readFileSync(process.argv[2], 'utf8'));
let single = 0;
for (const day of days) {
  const ledger = buildEpisodeLedger(day);
  if (!ledger.quiet.rows.length) continue;
  if (ledger.quiet.rows.length === 1) single += 1;
  const covered = ledger.findings.map((entry) => entry.row.t)
    .filter((t) => t > ledger.quiet.start && t < ledger.quiet.end);
  if (covered.length) {
    console.log(`${day.date}: Quiet ${ledger.quiet.rows.length}, span ${clock(ledger.quiet.start)}–${clock(ledger.quiet.end)}, covers Findings at ${covered.map(clock).join(', ')}`);
  }
}
console.log(`days with Quiet rows and exactly one quiet anchor: ${single}`);
